sources/core/Stride.Core.Mathematics/Size2.cs

```diff
@@ -66,6 +66,35 @@ public Size2(int width, int height)
         /// </summary>
         [DataMember(1)]
         public int Height;
+        
+        /// <summary>
+        /// Gets or sets the component at the specified index.
+        /// </summary>
+        /// <value>The value of the Width or Height component, depending on the index.</value>
+        /// <param name="index">The index of the component to access. Use 0 for the Width component and 1 for the Height component.</param>
+        /// <returns>The value of the component at the specified index.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">Thrown when the <paramref name="index"/> is out of the range [0, 1].</exception>
+        public int this[int index]
+        {
+            get
+            {
+                switch (index)
```

Comment:
You can use the switch pattern here. It has a nicer syntax.

Reply:
I totally agree, I was opting to keep the changes to the type as similar to the other Stride Mathmatics types as possible to keep the source code and usage consistent. I figured it was better to just update the mathmatics types all at once in another PR, then to have some using one convention and others using another.